Tasks/DotNetCoreCLIV2/dotnetcore.ts

```diff
@@ -325,22 +329,50 @@ export class dotNetExe {
         }
 
         var projectFiles = utility.getProjectFiles(projectPattern);
+        var resolvedProjectFiles: string[] = [];
 
         if (searchWebProjects) {
-            projectFiles = projectFiles.filter(function (file, index, files): boolean {
+            resolvedProjectFiles = projectFiles.filter(function (file, index, files): boolean {
                 var directory = path.dirname(file);
                 return tl.exist(path.join(directory, "web.config"))
                     || tl.exist(path.join(directory, "wwwroot"));
             });
 
-            if (!projectFiles.length) {
-                tl.error(tl.loc("noWebProjectFound"));
-            }
+            if (!resolvedProjectFiles.length) {
+                var projectFilesUsingWebSdk = projectFiles.filter(this.isWebSdkUsed)
+                if(!projectFilesUsingWebSdk.length) {
+                    tl.error(tl.loc("noWebProjectFound"));
+                }
+                return projectFilesUsingWebSdk;
+            } 
+            return resolvedProjectFiles;
         }
 
         return projectFiles;
     }
 
+    private isWebSdkUsed(projectfile: string) {
+        var fileBuffer: Buffer = fs.readFileSync(projectfile);
+        var fileEncodeType = fileEncoding.detectFileEncoding(projectfile, fileBuffer);
+        var webConfigContent: string = fileBuffer.toString(fileEncodeType[0]);
+
+        if(fileEncodeType[1]) {
+            webConfigContent = webConfigContent.slice(1);
+        }
+
+        var ltxDom = new ltxdomutility.LtxDomUtility(webConfigContent);
+        var projectDomArray: Array<any> = ltxDom.getElementsByTagName("project");
```

Comment:
instead of this you can just use ltx.parse(webConfigContent).getAttr("Sdk")

Reply:
Done